tools/lib/src/syncable_process_runner.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:file/file.dart';
+import 'package:flutter_plugin_tools/src/common/core.dart';
+import 'package:flutter_plugin_tools/src/common/process_runner.dart';
+
+/// A class used to run processes synchronously.
+///
+/// We extend [ProcessRunner] to stub behaviors in tests.
+class SyncableProcessRunner extends ProcessRunner {
```

Comment:
I've extended the class in case I needed to mock the `runSync` method as extension methods cannot be mocked, but I see that I haven't needed it in tests. I'll use the extension method.